app/models/mixins/retirement_mixin.rb

```diff
@@ -112,18 +112,19 @@ def raise_retire_audit_event(message)
 
   def retirement_check
     return if retired? || retiring? || retirement_initialized?
+    requester = system_context_requester
 
     if !retirement_warned? && retirement_warning_due?
       begin
         self.retirement_last_warn = Time.now.utc
         save
-        raise_retirement_event(retire_warn_event_name)
+        raise_retirement_event(retire_warn_event_name, requester)
       rescue => err
         _log.log_backtrace(err)
       end
     end
 
-    self.class.make_retire_request(self.id) if retirement_due?
+    self.class.make_retire_request(self.id, *requester) if retirement_due?
```

Comment:
Is `requester` an Array here?

Reply:
No, but it may or may not have a requester at that point.